src/components/Map/OpenLayersMap.jsx

```diff
@@ -0,0 +1,157 @@
+import React, { useRef, useEffect } from 'react';
+import * as ol from 'ol';
+import { ScaleLine, Zoom } from 'ol/control';
+import { MouseWheelZoom } from 'ol/interaction';
+import styles from './Map.module.css';
+import 'ol/ol.css';
+
+function OpenLayersMap({
+  setMapObj,
+  map,
+  children,
+  setMap,
+  zoom,
+  minZoom,
+  maxZoom,
+  center,
+  showScale,
+  scaleUnits,
+  scaleBar,
+  scaleBarPosition,
+  enableMouseWheelZoom,
+  enableZoomControls,
+  zoomControlsPosition,
+}) {
+  const mapRef = useRef();
+
+  // on component mount
+  useEffect(() => {
+    const options = {
+      view: new ol.View({
+        zoom,
+        center,
+        minZoom,
+        maxZoom,
+      }),
+      layers: [],
+      overlays: [],
+      controls: [],
+    };
+
+    const mapObject = new ol.Map(options);
+
+    if (showScale) {
+      let scaleClassName = 'ol-scale-line';
+      if (scaleBar) scaleClassName = 'ol-scale-bar';
+      const scalePosClass = `scalePos${scaleBarPosition}`;
+      scaleClassName = `${styles[scalePosClass]} ${scaleClassName}`;
+      const control = new ScaleLine({
+        units: scaleUnits,
+        bar: scaleBar,
+        className: scaleClassName,
+        minWidth: 100,
+      });
+      mapObject.addControl(control);
+    }
+
+    if (enableMouseWheelZoom) {
+      mapObject.getInteractions().forEach((interaction) => {
+        if (interaction instanceof MouseWheelZoom) {
+          interaction.setActive(true);
+        }
+      }, this);
+    } else {
+      mapObject.getInteractions().forEach((interaction) => {
+        if (interaction instanceof MouseWheelZoom) {
+          interaction.setActive(false);
+        }
+      }, this);
+    }
+
+    if (enableZoomControls) {
+      const zoomClass = `${styles.ol - zoom} ${styles[`POS-${zoomControlsPosition}`]}`;
+      mapObject.addControl(new Zoom({ delta: 0.3, className: zoomClass }));
+    }
+
+    mapObject.setTarget(mapRef.current);
+    if (setMap) setMap(mapObject);
+    if (setMapObj) setMapObj(mapObject);
```

Comment:
Do we have a difference between setMap and setMapObj?

Reply:
yes there is a difference, depending on the scope: Map Vizzard including interactive options panel vs standalone Map created from JSON configuration.